packages/stepper/src/Stepper.tsx

```diff
@@ -0,0 +1,521 @@
+    .${stepBoxTopStyle}:before {
+      transform: skewX(${Math.atan(2 / 3)}rad);
+    }
+
+    .${stepBoxBottomStyle}:before {
+      transform: skewX(-${Math.atan(2 / 3)}rad);
+    }
+  }
+`;
+
+const upcomingStepStyle = css`
+  color: ${uiColors.gray.base};
+  z-index: 1;
+
+  &:last-of-type {
+    z-index: 0;
+  }
+
+  .${stepBoxTopStyle}, .${stepBoxBottomStyle} {
+    & > *,
+    &:before,
+    &:after {
+      background: ${uiColors.gray.light3};
+      border-color: ${uiColors.gray.light2};
+    }
+  }
+`;
+
+const hiddenLayerStyle = css`
+  opacity: 0;
+  visibility: hidden;
+  pointer-events: none;
+`;
+
+const slideInAnimationStyle = css`
+  left: 0;
+  transition: left 700ms;
+`;
+
+const slideOutAnimationStyle = css`
+  left: -30px;
+  transition: left 700ms;
+`;
+
+const containerStyle = css`
+  position: relative;
+  height: 46px;
+  width: 100%;
+`;
+
+const stepLabelStyle = css`
+  display: flex;
+  position: absolute;
+
+  align-items: center;
+  justify-content: center;
+  top: calc(100% - 12px);
+
+  height: 24px;
+  min-width: 24px;
+  max-width: max-content;
+
+  font-size: 12px;
+  line-height: 14px;
+
+  border-radius: 12px;
+  border: 1px solid;
+`;
+
+const completedStepLabelStyle = css`
+  background-color: white;
+  border-color: ${uiColors.green.base};
+  color: ${uiColors.green.base};
+`;
+
+const currentStepLabelStyle = css`
+  background-color: ${uiColors.gray.dark2};
+  border-color: #464c4f;
+  color: white;
+`;
+
+const upcomingStepLabelStyle = css`
+  background-color: white;
+  border-color: ${uiColors.gray.light1};
+  color: ${uiColors.gray.base};
+`;
+
+const stepLabelTextStyle = css('padding: 0 8px;');
+
+function InternalStep({
+  children,
+  state,
+  stepLabel,
+  className,
+  ...containerProps
+}: InternalStepProps) {
+  const isCurrent = state === 'current';
+  const isUpcoming = state === 'upcoming';
+  const isCompleted = state === 'completed';
+  return (
+    <div
+      className={cx(
+        stepStyle,
+        { [upcomingStepStyle]: isUpcoming },
+        { [currentStepStyle]: isCurrent },
+        className,
+      )}
+      {...containerProps}
+    >
+      {stepLabel && (
+        <div
+          className={cx(
+            stepLabelStyle,
+            state === 'current'
+              ? currentStepLabelStyle
+              : isCompleted
+              ? completedStepLabelStyle
+              : upcomingStepLabelStyle,
+          )}
+        >
+          {isCompleted ? (
+            <CheckmarkIcon />
+          ) : (
+            <div className={stepLabelTextStyle}>{stepLabel}</div>
+          )}
+        </div>
+      )}
+      {children}
+      <div className={stepBoxStyle}>
+        <span className={stepBoxTopStyle}>
+          <div />
+        </span>
+        <span className={stepBoxBottomStyle}>
+          <div />
+        </span>
+      </div>
+    </div>
+  );
+}
+
+InternalStep.displayName = 'Step';
+
+type StepContents = string | number | Array<StepContents>;
```

Comment:
Feels like we should break this into two types, `StepContent` and `StepContents` for the array instance. Otherwise as you have the `Step` component typed now doing `<Step>{[1,2,3]}</Step>` is allowed.

Reply:
Yep, that's by design. For example, `<Step>Step {index}<Step>` is equivalent to `<Step>{['Step ', 1]}</Step>`. This lends to slightly more convenience than needing to do ``<Step>{`Step ${index}`}<Step>``